mobile/android/logcat-helper.ts

```diff
@@ -3,30 +3,31 @@ import { DeviceAndroidDebugBridge } from "./device-android-debug-bridge";
 
 export class LogcatHelper implements Mobile.ILogcatHelper {
 	private mapDeviceToLoggingStarted: IDictionary<boolean>;
+	private mapDeviceToLogcat:any;
+	private mapDeviceToLineStream:any;
 
 	constructor(private $deviceLogProvider: Mobile.IDeviceLogProvider,
 		private $devicePlatformsConstants: Mobile.IDevicePlatformsConstants,
 		private $logger: ILogger,
 		private $injector: IInjector,
 		private $processService: IProcessService) {
 		this.mapDeviceToLoggingStarted = Object.create(null);
+		this.mapDeviceToLogcat = Object.create(null);
```

Comment:
Instead of creating three different dictionaries with the same key (`deviceIdentifier`), you can create a single one with two properties:
```TypeScript
interface IDeviceLoggingData {
    loggingProcess: ChildProcess; // (or any in case you define the interface in .d.ts)
    lineStream: any;
}

private devicesLoggingData: IDicitionary<ILoggingData>;
```

This way in `start` method you can use:
```TypeScript
const deviceLoggingData = this.devicesLoggingData[deviceIdentifier];
if (deviceIdentifier && !deviceLoggingData) {
...
}
```

And in the `stop` method, after removing all listeners, just delete the entry:
```
delete this.devicesLoggingData[deviceIdentifier];
```

The benefits of this solution are:
1. No need of multiple dictionaries - A single data object containing all the information.
2. No need to set true/false values, just check if there's an entry in the dictionary.
3. The code in start method will be easier for reading as it will use a single object.

What do you think?

Reply:
Basically, I want to keep it simple, so I implemented the simplest solution, but I have no preference.

Reply:
You can create the interface in the same file and use it only here. This way it will not be visible outside of the file.